File: 3-LongestSubString.js
function lengthOfLongestSubstring(s) {
    let max = 0; 
    let start = 0;
    let check = new Set();
    for (let i = 0; i < s.length; i++) {
        for (let j = start; j < s.length; j++) {
            if (!check.has(s[j])) {
                check.add(s[j]);
                if (max < (j - i + 1)) {
                    max = j - i + 1;
                }
            } else {
                start = j;
                check.delete(s[i]);
                break;
            }
        }
    }
    return max;
}
console.log(lengthOfLongestSubstring("abcabcbb"))

File: 3-LongestSubString.py
def lengthOfLongestSubstring(s):
    max = 0
    start = 0
    chars = set()
    for i in range(0, len(s)):
        for j in range(start, len(s)): 
            if (s[j] not in chars): 
                chars.add(s[j])
                if max < (j - i + 1): 
                    max = j - i + 1
            else: 
                start = j
                chars.remove(s[i])
                break
    return max 

print(lengthOfLongestSubstring('abcabcbb'))
        